saleor/search/search_indexes.py

```diff
@@ -0,0 +1,54 @@
+from haystack import indexes
+from ..product.models import Product
+from ..order.models import Order
+from ..userprofile.models import User
+
+
+class ProductIndex(indexes.SearchIndex, indexes.Indexable):
+    name = indexes.CharField(model_attr='name')
+    text = indexes.MultiValueField(document=True)
+    categories = indexes.MultiValueField()
+
+    def get_model(self):
+        return Product
+
+    def index_queryset(self, using=None):
+        qs = self.get_model().objects.get_available_products()
```

Comment:
We need to index all products and reimplement availability checks at search time (but only when searching the site and not the dashboard).